vms/platformvm/service.go

```diff
@@ -1276,17 +1291,10 @@ func (s *Service) AddValidator(req *http.Request, args *AddValidatorArgs, reply
 		changeAddr,
 	)
 	if err != nil {
-		return fmt.Errorf("couldn't create tx: %w", err)
+		return nil, ids.ShortEmpty, fmt.Errorf("couldn't create tx: %w", err)
 	}
 
-	reply.TxID = tx.ID()
-	reply.ChangeAddr, err = s.addrManager.FormatLocalAddress(changeAddr)
-
-	return utils.Err(
-		err,
-		s.vm.Network.IssueTx(req.Context(), tx),
-		user.Close(),
-	)
+	return tx, changeAddr, nil
```

Comment:
Should this call `user.Close`? I don't _really_ have a strong opinion here (we went back and forth on this when we first wrote this code). The reason we previously called `user.Close()` in the defer _and_ at the end of the function was so that:
- If this function was already erroring, `user.Close()` would be called and any error thrown away.
- If this function was attempting to return successfully, `user.Close()` would be called and any error returned.

Reply:
Yeah we should add the close call here.